### PR TITLE
Do not trigger deprecation error in ResolveParameterPlaceHoldersPass

### PR DESCRIPTION
**Pull request by murnieza**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | none |
| License | MIT |
| Doc PR | none |

### CODE REVIEW

File: src/Symfony/Component/DependencyInjection/Compiler/ResolveParameterPlaceHoldersPass.php [hunk 1]
```diff
@@ -38,7 +38,7 @@ public function process(ContainerBuilder $container)
                 $definition->setFile($parameterBag->resolveValue($definition->getFile()));
                 $definition->setArguments($parameterBag->resolveValue($definition->getArguments()));
                 if ($definition->getFactoryClass(false)) {
-                    $definition->setFactoryClass($parameterBag->resolveValue($definition->getFactoryClass()));
+                    $definition->setFactoryClass($parameterBag->resolveValue($definition->getFactoryClass(false)));
```
Comment (iltar) [inline src/Symfony/Component/DependencyInjection/Compiler/ResolveParameterPlaceHoldersPass.php]: I think this is no longer required, see #14900.

Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/Compiler/ResolveParameterPlaceHoldersPass.php]: Yes it is! We still have to trigger the relevant notices, and only them. Silencing can be turned off (and will be by users preparing for 3.0 migration).

**Comment by nicolas-grekas** [comment]
:+1: should be merged on 2.7

**Comment by nicolas-grekas** [comment]
I voted to quickly, removing it for now. When this code path is taken, it means one is using a deprecated service factories configuration. Why should we hide the notice? To me, there is something to upgrade, thus to warn about. Did I miss something?

**Comment by stof** [comment]
@nicolas-grekas but you already got the notice in the place using it. This will trigger a second notice in a place out of the control of the user

**Comment by stof** [comment]
btw, the call to ``setFactoryClass`` will also trigger a deprecation warning here, so you would even get 2 new warnings

**Comment by nicolas-grekas** [comment]
@stof true, thanks for the review! :+1: then

**Comment by fabpot** [comment]
Thank you @murnieza.

**Pull request closed without merging**